src/main/webapp/src/app/my_gardens_page/my-gardens.component.ts

```diff
@@ -25,5 +66,291 @@ export class MyGardensComponent implements OnInit {
+   */
+  getUserInfo(user: string): Observable<HttpResponse<User>> {
+    return this.httpClient.get<User>('/user/' + user, {
+      observe: 'response',
+      responseType: 'json',
+    });
+  }
+
+  /**
+   * Obtains a garden's administrator name, and inserts it into
+   * global set.
+   *
+   * @param adminId the admin id.
+   */
+  getGardenAdminName(adminId: string) {
+    this.gardenAdminNames.set(adminId, 'Loading...');
+    // Obtain user names
+    this.getUserInfo(adminId).subscribe({
+      next: (response: HttpResponse<User>) => {
+        // Successful responses are handled here.
+        this.gardenAdminNames.set(adminId, response.body.preferredName);
+      },
+      error: (error: HttpErrorResponse) => {
+        // Handle connection error
+        if (error.error instanceof ErrorEvent) {
+          console.error('Network error: ' + error.error.message);
+          this.gardenAdminNames.set(adminId, 'Cannot fetch name');
+          return;
+        }
+        console.error('Unexpected error: ' + error.statusText);
+        this.gardenAdminNames.set(adminId, 'Cannot fetch name');
+      },
+    });
+  }
+
+  /**
+   * Deletes a garden from a user's garden list.
+   *
+   * Performs DELETE: /user/current/garden-list/{id}
+   *
+   * @param id the garden id to delete.
+   * @return the http response.
+   */
+  deleteFromUserGardenList(id: string): Observable<HttpResponse<string>> {
+    return this.httpClient.delete<string>('/user/current/garden-list/' + id, {
+      observe: 'response',
+      responseType: 'json',
+    });
+  }
+
+  /**
+   * Constructs a user garden admin list.
+   * Step 1 in construction.
+   *
+   */
+  createUserGardenAdminList(): void {
+    this.getUserGardenAdminList().subscribe({
+      next: (response: HttpResponse<Array<string>>) => {
+        // Successful responses are handled here.
+        this.myGardenAdminIdList = response.body;
+        // Gets user names
+        this.myGardenAdminMap = new Map<string, Garden>();
+        // Initiates next step
+        this.createUserGardenList();
+        this.myGardenAdminIdList.forEach(id => {
+          this.getGardenInfo(id).subscribe({
+            // Obtain user names
+            next: (response: HttpResponse<Garden>) => {
+              // Successful responses are handled here.
+              this.myGardenAdminMap.set(id, response.body);
+              this.getGardenAdminName(response.body.adminId);
+            },
+            error: (error: HttpErrorResponse) => {
+              // Handle connection error
+              if (error.error instanceof ErrorEvent) {
+                console.error('Network error: ' + error.error.message);
+                return;
+              }
+              console.error('Unexpected error: ' + error.statusText);
+            },
+          });
+        });
+      },
+      error: (error: HttpErrorResponse) => {
+        // Handle connection error
+        if (error.error instanceof ErrorEvent) {
```

Comment:
I believe [this comment](https://github.com/googleinterns/step26-2020/pull/41#discussion_r464416270) applies here as well

Reply:
Missed that -- Thanks for pointing out!